### ISSUE TITLE
Add ability to download uploaded markdown

### ISSUE THREAD

**Issue by DavidSchinazi** [ietf-tools/datatracker#6599]
### Description

Thanks for adding the ability to upload agendas in markdown, it's amazing. However, I've noticed that when we upload markdown, I'm only able to download the rendered HTML (that's based on the markdown) but not the markdown itself. It makes collaborating between co-chairs more difficult because if one chair uploads a -00 in markdown and the co-chair wants to edit it to create a -01, they have to email the first chair to get the markdown.

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by pusateri** [comment]
This used to work when I added the feature. I tested it with the following command and got markdown. Seems like now it's broken.
```
curl --header "Accept: text/markdown" https://datatracker.ietf.org/meeting/118/materials/agenda-118-dnssd-02.md
```

But it still works if you remove the .md extension. It should work with the .md extension too.
```
curl --header "Accept: text/markdown" https://datatracker.ietf.org/meeting/118/materials/agenda-118-dnssd-02
```

**Comment by pusateri** [comment]
It seems now the opposite is true. Asking for .md gives markdown as expected and no suffix gives html.

**Comment by richsalz** [comment]
Yes, the default is HTML. If you want the markdown, you have to do *both* the `Accept` header field, and the `.md` extension. This is problematic for browser-based clients which do not have a convenient way to set the header.  Looking to see if this is a datatracker bug or an IETF CDN issue.

**Comment by richsalz** [comment]
Even in localhost/dev mode (e.g., view-source:http://localhost:8000/meeting/122/materials/agenda-122-httpapi-00.md) it outputs html, so seems like a datatracker bug.

**Comment by richsalz** [comment]
(Deleted some previous, confused, comments)

All six of these commands do the right thing, giving me markdown:
```
curl -D- https://datatracker.ietf.org/meeting/122/materials/agenda-122-httpapi-00.md
curl -D- http://localhost:8000/meeting/122/materials/agenda-122-httpapi-00.md
curl -D- --header "Accept: text/markdown" https://datatracker.ietf.org/meeting/122/materials/agenda-122-httpapi-00
curl -D- --header "Accept: text/markdown" http://localhost:8000/meeting/122/materials/agenda-122-httpapi-00
curl -D- --header "Accept: text/markdown" https://datatracker.ietf.org/meeting/122/materials/agenda-122-httpapi-00.md
curl -D- --header "Accept: text/markdown" http://localhost:8000/meeting/122/materials/agenda-122-httpapi-00.md
```
So local or remote, adding *either* the HTTP `Accept` field or the `.md` extension gets the markdown format. Maybe the browser is auto-converting for display?

Adding a browser into the exchange (Firefox on Mac):

- If I go to `https://datatracker.ietf.org/doc/agenda-122-httpapi/` and right-click on the markdown link (which is `https://datatracker.ietf.org/meeting/122/materials/agenda-122-httpapi-00.md`) and then "save as" I get markdown in the saved file. **So right-click and save, seems to do the right thing.**
- If  I go to `http://localhost:8000/doc/agenda-122-httpapi/` right-click on the markdown link (which is `https://datatracker.ietf.org/meeting/122/materials/agenda-122-httpapi-00.md` *interesting*) also gives me markdown. To me that emphasizes that the browser is doing the HTML rendering.

If I go to the `.md` of the agenda, Chrome and Safari display markdown if the URL is https to datatracker, and render into HTML if the URL is http to localhost.

Based on all this, it seems to me the thing to do is change the other versions row to add a "Download/save" suggestion.

**Comment by jennifer-richards** [comment]
I think it's the accept header, not the browser rendering markdown. Also using FF on Mac, the request that's sent when you left click or directly access the URL includes
```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
```
in the request headers. The response is HTML that matches what `curl` does if you pass that accept header.

The download/save is a great find! IMO that's the right thing to do rather than providing an endpoint that ignores the accept header.

**Comment by richsalz** [comment]
> The download/save is a great find! IMO that's the right thing to do rather than providing an endpoint that ignores the accept header.

So I should add the download/save text as mentioned above?

**Comment by jennifer-richards** [comment]
Sounds good to me

**Comment by jennifer-richards** [comment]
See also the report/discussion in #9268, which is the same underlying issue.

Particularly relevant to the download/save approach is the not-great observation:

> However, after downloading that way, I also got the raw markdown source just by clicking on the "markdown" link. And a few minutes later, both methods are back to serving me HTML. It may be that our edge caches are interfering with the Accept / Content-type headers. We'll investigate.

**Comment by jennifer-richards** [comment]
After a quick review, I don't think CF caching can easily interact with content-type, so the "Save as" download mechanism won't work reliably.

I think we'll need to adjust the endpoint to ignore the Accept header and return the content type appropriate to the file extension.

**Comment by dkg** [comment]
sad that IETF technology (the `Accept` header) doesn't work for IETF infrastructure ☹

(edited to add: of course, as seen in #9268 i can't easily tell my browser to send a different `Accept` header anyway)

**Comment by jennifer-richards** [comment]
> sad that IETF technology (the `Accept` header) doesn't work for IETF infrastructure ☹

Edge caches and server-driven content negotiation are tricky to coordinate. I'm pretty sure it can be done with our infrastructure, but would probably need to be configured per-endpoint or per-family-of-endpoints where it's needed.

In this case, I'm not sure that it'd be the right solution anyway. If you click on the big shiny "markdown" button, giving you the markdown source is probably the right thing to do. For the extensionless URL, the accept header might be nice, but I think it's reasonable for it always to return a particular format (i.e., html unless I've got myself confused).

**Comment by jennifer-richards** [comment]
Possibly helpful CF feature: https://developers.cloudflare.com/cache/advanced-configuration/serve-tailored-content/#example-same-url-different-content

**Comment by rjsparks** [comment]
Interesting, but I think we need to hit this one with a blunt instrument and have a separate path for markdown source vs rendered html.
